vulnerabilities/pipelines/huawei_bulletin_importer.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+import logging
+from datetime import datetime
+from datetime import timezone
+from typing import Iterable
+from typing import Tuple
+
+import requests
+from bs4 import BeautifulSoup
+from packageurl import PackageURL
+from univers.version_range import VersionRange
+from univers.versions import GenericVersion
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipeline
+from vulnerabilities.severity_systems import GENERIC
+
+logger = logging.getLogger(__name__)
+
+
+def extract_version(version_str: str) -> Tuple[str, str]:
+    """
+    Gets the version type and number from a messy Huawei version string.
+    Returns tuple of (type, version) - returns empty type if cant parse
+    """
+    version_str = version_str.lower().strip()
+    if version_str.startswith("harmonyos"):
+        return "harmonyos", version_str.split("harmonyos")[-1].strip()
+    elif version_str.startswith("emui"):
+        return "emui", version_str.split("emui")[-1].strip()
+    return "", version_str
+
+
+class HuaweiBulletinImporterPipeline(VulnerableCodeBaseImporterPipeline):
+    """
+    Scrapes security bullitin data from Huawei's website:
+    https://consumer.huawei.com/en/support/bulletin/
+    """
+
+    pipeline_id = "huawei_bulletin_importer"
+    spdx_license_expression = "LicenseRef-Terms-Of-Use"
+    license_url = "https://consumer.huawei.com/en/legal/terms-of-use/"
+    url = "https://consumer.huawei.com/en/support/bulletin/"
+    importer_name = "Huawei Bulletin Importer"
+
+    def __init__(self):
+        super().__init__()
+        self.raw_data = None
+
+    @classmethod
+    def steps(cls):
+        """The steps we need to run in order"""
+        return (
+            cls.fetch_bulletin,
+            cls.collect_and_store_advisories,
+            cls.import_new_advisories,
+        )
+
+    def fetch_bulletin(self):
+        """
+        Gets the bullitin data from Huawei's site.
+        Stores raw html for processing later.
+        """
+        self.log(f"Fetching {self.url}")
+        try:
+            response = requests.get(f"{self.url}2024/9/")
```

Comment:
It does not make sense to collect advisories only from Sep-2024, we should collect all the security bulletins from https://consumer.huawei.com/en/support/bulletin/.